text/0152-telemetry-schemas.md

```diff
@@ -0,0 +1,1159 @@
+implementations will no longer work correctly together.
+
+Essentially there is a coupling between 3 parties: 1) OpenTelemetry semantic
+conventions, 2) telemetry sources and 3) telemetry consumers. The coupling
+complicates the independent evolution of these 3 parties.
+
+We recognize the following needs:
+
+- OpenTelemetry semantic conventions need to evolve over time. When conventions
+  are first defined, mistakes are possible and we may want to fix the mistakes
+  over time. We may also want to change conventions to re-group the attributes
+  into different namespaces as our understanding of the attribute taxonomy
+  improves.
+
+- Telemetry sources over time may want to change the schema of the telemetry
+  they emit. This may be because for example the semantic conventions evolved
+  and we want to make our telemetry match the newly introduced conventions.
+
+- In an observability system there may simultaneously exist telemetry sources
+  that produce data that conforms to different telemetry schemas because
+  different sources evolve at a different pace and are implemented and
+  controlled by different entities.
+
+- Telemetry consumers have a need to understand what schema a particular piece
+  of received telemetry confirms to. The consumers also need a way to be able to
+  interpret the telemetry data that uses different telemetry schemas.
+
+This document proposes a solution to these needs.
+
+## Solution Summary
+
+We believe that the 3 parties described above should be able to evolve
+independently over time, while continuously retaining the ability to correctly
+work together.
+
+Telemetry Schemas are central to how we make this possible. Here is a summary of
+the proposal:
+
+- We introduce a file format for defining telemetry schemas.
+
+- Telemetry schemas are versioned. Over time the schema may evolve and telemetry
+  sources may emit data confirming to newer versions of the schema.
+
+- Telemetry schemas explicitly define transformations that are necessary to
+  convert telemetry data between different versions of the schema, provided that
+  such conversions are possible. When conversions are not possible it
+  constitutes a breaking change between versions.
+
+- Telemetry schemas are identified by Schema URLs.
+
+- Telemetry sources (e.g. instrumentation libraries) should include a schema URL
+  in the emitted telemetry.
+
+- Telemetry consumers should pay attention to the schema of the received
+  telemetry. If necessary, telemetry consumers may transform the telemetry data
+  from the received schema version to the target schema version as expected at
+  the point of use (e.g. a dashboard may define which schema version it
+  expects).
+
+- OpenTelemetry will publish a telemetry schema as part of the specification.
+  The schema will contain the list of transformations that semantic conventions
+  undergo. The schema will be available to be referred and downloaded at well
```

Comment:
Should the "known URL" be the URL of the schema?

Reply:
Yes, the doc later explains what a Schema URL is in more details.